includes/enrolment/class-sensei-enrolment-provider-state-store.php

```diff
@@ -223,4 +220,61 @@ public static function persist_all() {
 	private static function get_providers_state_meta_key( $course_id ) {
 		return self::META_PREFIX_ENROLMENT_PROVIDERS_STATE . $course_id;
 	}
+
+	/**
+	 * Register a possible update in the status of a provider for a user and a course. If there was no actual change on
+	 * the status, this method has no effect.
+	 *
+	 * @param array $provider_results An array with the format 'provider_id' => enrollment result.
+	 * @param int   $user_id          The user which the change applies to.
+	 * @param int   $course_id        The course which the change applies to.
+	 */
+	public static function register_possible_enrolment_change( $provider_results, $user_id, $course_id ) {
+		$state_store = self::get( $user_id, $course_id );
+
+		$new_snapshot = Sensei_Enrolment_Provider_State_Snapshot::duplicate( $state_store->get_current_snapshot() );
+		$has_changed  = false;
+
+		foreach ( $provider_results as $provider_id => $is_enrolled ) {
```

Comment:
It looks like if a provider drops off (for example: product removed from a course), it's previous state will continue on in the snapshot history.

Reply:
Good catch! 21310921ac1fd7abe0ae6042cabbb86ec5b01555